app/Http/Livewire/Backend/CreateCourses.php

```diff
@@ -140,6 +144,26 @@ protected function validateCurrentStep()
     }
 }
 
+protected function validateMarksAllocation()
+{
+    // Check if at least one field is set (not null or empty)
+    if (!empty($this->marks_allocation['practicals']) || 
+        !empty($this->marks_allocation['project']) || 
+        !empty($this->marks_allocation['mid_exam']) || 
+        !empty($this->marks_allocation['end_exam'])) {
+        
+        // Cast all values to integers (treat empty values as 0)
+        $totalMarks = (int) ($this->marks_allocation['practicals'] ?? 0) +
+                      (int) ($this->marks_allocation['project'] ?? 0) +
+                      (int) ($this->marks_allocation['mid_exam'] ?? 0) +
+                      (int) ($this->marks_allocation['end_exam'] ?? 0);
+
+        // Check if total is exactly 100
+        if ($totalMarks != 100) {
+            $this->addError('marks_allocation.total', 'The total of marks allocation must be 100.');
+        }
+    }
+}
```

Comment:
Can you try to implement this by using a for loop? It might be more suitable in terms of adding a new mark allocation option in future

Reply:
I will create a for loop